backend/src/campaigns.rs

```diff
@@ -63,6 +62,26 @@ pub async fn create_or_update_campaign(
     Ok(Json(()))
 }
 
+#[delete("/<campaign_id>")]
+pub async fn delete_campaign(
+    campaign_id: i32,
+    user: User,
+    db: Database,
+) -> Result<Json<()>, Json<CampaignError>> {
+    let admin_res = db
+        .run(move |conn| OrganisationAdmin::new_from_campaign_id(user, campaign_id, conn))
+        .await;
```

Comment:
`.ok_or(Json(CampaignError::Unauthorized))?;`

unfold below match into the Ok branch